### ISSUE TITLE
Metadata translator - Synonyms

### ISSUE THREAD

**Issue by benito2313** [microsoft/Analysis-Services#100]
Hi,

Is it possible to also translate the synonyms?
Because when its connected to a powerbi File or AS, i cant see the synonyms that need to be translated.

Kind regards,

Ben

**Comment by KayUnkroth** [comment]
Yes, I think that's doable. It's a bit of a bigger change, but I'll add it to the backlog. Might take a few weeks.

**Comment by otykier** [comment]
@KayUnkroth on that note, I would love to have some kind of documentation for the synonym spec (all we know from the public documentation is that it's a [JSON or XML string](https://docs.microsoft.com/en-us/dotnet/api/microsoft.analysisservices.tabular.linguisticmetadata.content?view=analysisservices-dotnet#Microsoft_AnalysisServices_Tabular_LinguisticMetadata_Content)). Many users want to add/edit synonyms through Tabular Editor, but before I provide UI for that, I'd like to know the specs.

**Comment by marcosqlbi** [comment]
@otykier there is a yaml file [documented here](https://docs.microsoft.com/en-us/power-bi/natural-language/q-and-a-tooling-advanced).
But I don't know how it's used in the TOM properties. I don't think [this blog post](https://data-marc.com/2020/02/10/synonyms-the-secret-behind-power-bi-qa-and-natural-language-query/) can help.

**Comment by otykier** [comment]
@marcosqlbi ideally the YAML format can be converted 1:1 to the JSON or XML string in the Content property of TOM that I linked above, but I haven't seen any documentation stating that this is indeed the case.

**Comment by KayUnkroth** [comment]
Daniel, Marco, this request here is about translating synonyms. I believe I can do that with pure TOM. We'll see how it goes. Better documentation for synonyms? You might want to ping Will and/or Marius via email. I'm sure they will help you.

**Comment by marcosqlbi** [comment]
@KayUnkroth I just wanted to help @benito2313 - I don't have plans to work on synonyms or linguistic schema soon!

**Comment by benito2313** [comment]
Hi all,

I also have a request at the tabular editor GitHub @otykier https://github.com/otykier/TabularEditor/issues/257 
It’s closed now, but I added an comment a few days ago.

Because I would like to add synonyms to the model through tabular editor and then translate them with the metadata translator. Well this is the vision I have 😊😬

Kind regards,

Ben

**Comment by KayUnkroth** [comment]
Awesome! The pressure is on you, Daniel! :-) And the race is on. Let's see who has it first, TE editing or MT translating. :-)

**Comment by otykier** [comment]
@KayUnkroth done: https://github.com/otykier/TabularEditor/releases/tag/2.15.2 

I spent waaaaay too long time trying to make sense of the Linguistic Schema Definition Language. Just an FYI: The documentation that Marco linked is outdated. I only figured this out when I had already spent hours trying to create contract classes based on the official linguistic schema spec... *sigh*

**Comment by aaron-meyers** [comment]
Hey everyone, jumping in the fun here - I'm the engineering manager for Q&A in Power BI. Kay mentioned this thread to me.

We have a JSON schema for the JSON LSDL that goes in TOM LinguisticMetadata (the XML variant is deprecated and obsolete). You can get it by going to aka.ms/lsdlschema or json.schemastore.org/lsdlschema - this links to the master schema which has references to specific versions, e.g. the latest published version is at https://json.schemastore.org/lsdlschema-3.0.json. If you're using the latest Power BI Desktop you'll notice that it's actually using 3.1 version and we have an outstanding work item to publish that version to schemastore.

That said, we honestly haven't heard a lot of interest from customers in working with linguistic schema directly so we've been focusing on improvements to our Q&A setup UI within PBI Desktop for modifying the linguistic schema.

As for translating synonyms, since Q&A is basically English-only today (*with a very limited Spanish preview that we haven't updated recently) there's not much value in linguistic schema translations. The Q&A engine in PBI Desktop and PBI web will only actually load linguistic schema from the default model culture currently.

More language support for Q&A, likely to start with Spanish and go from there, is something I'd love to do. Interested to hear more about your scenario @benito2313

**Comment by marcosqlbi** [comment]
@aaron-meyers For enterprise-level models, Power BI Desktop does not have a good UI (it does not access many features) and once a PBIX model is published and edited through XMLA endpoint, it can no longer edited in PBI Desktop.
Therefore, Tabular Editor is the only way to maintain such models.
Because investing on a semantic model is expensive, it could be an investment on datasets shared in many reports - probably datasets maintained with Tabular Editor rather than PBI Desktop.
Yes, there is not much interest today (als9 because of the complexity described above. By lowering the friction introducing LSDL schema editing in Tabular Editor and by improving documentation, you might see improvement in adoption.

**Comment by benito2313** [comment]
@aaron-meyers exactly what @marcosqlbi said.

We work with a AS model and connect to it in PBI for reporting but we work on the model in Tabular Editor.
Within our company there are many people with a lot of different languages so therefore it would be great to automate the translation of our model with the Metadata translator and breach the language barrier.
@aaron-meyers I have sent a request on LinkedIn so I can elaborate on what we are doing.

**Comment by benito2313** [comment]
Hi,

Is there any update about this?

Kind regards,

Ben

**Comment by KayUnkroth** [comment]
No updates, Ben. It's probably best to keep an eye out on the Power BI roadmap to see if/when additional language support is going to be added to Q&A. Until this happens, there is no point in adding support for synonym translations to Metadata Translator. The roadmap can be found here: https://docs.microsoft.com/en-us/power-platform-release-plan/2021wave1/power-bi/planned-features. 
Hope this helps.
Cheers,
Kay
